src/gitlab/functions/e2e-create-pipeline-enhancements.ts

```diff
@@ -95,6 +94,46 @@ export function enhanceE2eCreatePipelineScreen() {
 
         $('#MONAR_E2E_VARIABLES_BUTTONS').append(buttonHtml);
     });
+
+    // After the branch is changed, wait for the re-render and set default variable values again
+    const listOfBranches = $('fieldset:eq(0) ul')[0];
+    listOfBranches.addEventListener('click', () => setTimeout(fillAutomaticVariables, DelayAfterBranchChanged));
+}
+
+function fillAutomaticVariables(suiteButtons?: SuiteButton[]) {
+    const hashVariables = getHashVariables();
+    let formVariables = [];
+
+    // Pre-fill FE/BE versions
+    const versions = [];
+    if (hashVariables.fe_version) {
+        versions.push(`frontend_version: ${hashVariables.fe_version}`);
+    }
+    if (hashVariables.be_version) {
+        versions.push(`backend_version: ${hashVariables.be_version}`);
+    }
+    if (versions) {
+        formVariables.push(['ENVIRONMENT', versions.join('\\n')]);
+    }
+
+    // Project id and pipeline url of triggering project
+    if (hashVariables.source_project_id) {
+        formVariables.push(['SOURCE_PROJECT_ID', hashVariables.source_project_id]);
+    }
+    if (hashVariables.source_pipeline_url) {
+        formVariables.push(['SOURCE_PIPELINE_URL', hashVariables.source_pipeline_url]);
+    }
+
+    formVariables.reduce(
+        (previousVarInserted, currentVar) =>
+            new Promise(resolve =>
+                previousVarInserted.then(() => {
+                    insertNewVariable(currentVar[0] as string, currentVar[1] as string);
+                    setTimeout(resolve, DelayAfterAddingVariable);
```

Comment:
Again, maybe we could assert element being added using JQuery and length of element rather than using static time.

Reply:
It was my original intention to assert the state of the form, after each change that the code makes. But that would require rewriting the whole script - so that there are no mixed approaches, and the other stuff with buttons would then also work correctly. But it seemed too much work and out of scope.
Also, since gitlab UI is doing its async stuff and we do not know when it is finished, such assertion would need to be transparend and require some loop of wait->check->wait->check..., so there would still be some hard-coded waiting periods...

Reply:
Alright, I agree. Let's leave it for now and see how things work out.